src/input.rs

```diff
@@ -53,12 +53,18 @@ pub fn update_joystick<S: Hash + Sync + Send + Clone + Default + Reflect + 'stat
                 // Dragging
                 TouchPhase::Moved => {
                     if is_some_and(knob.id_drag, |i| i == *id) {
+                        knob.current_pos = *pos;
+                        let half = knob.interactable_zone_rect.half_size();
                         if node.behaviour == VirtualJoystickType::Dynamic {
                             knob.base_pos = *pos;
+                            let to_knob = knob.current_pos - knob.start_pos;
+                            let distance_to_knob = to_knob.length();
+                            if distance_to_knob > half.x {
+                                let excess_distance = distance_to_knob - half.x;
+                                knob.start_pos += to_knob.normalize() * excess_distance;
+                            }
```

Comment:
it's becoming "a lot" of code duplicated so maybe creating a function could help.

Reply:
I think it's a great idea, I was thinking of doing something similar, do you think you can do it in another PR, as soon as I accept this one?